### ISSUE TITLE
Incorrect REFL018 error when there is both an explicit and an internal member match

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/ReflectionAnalyzers#96]
Version: 0.1.7.0-dev

In this example, I want to invoke `Foo.Bar`, not `((IFoo)Foo).Bar`:

```c#
public interface IFoo
{
    event EventHandler Bar;
}

public class Foo : IFoo
{
    internal event EventHandler Bar;

    event EventHandler IFoo.Bar
    {
        add { }
        remove { }
    }
}
```

This is the only correct way to do so, so I expect no error:

```c#
typeof(Foo).GetEvent("Bar", BindingFlags.NonPublic | BindingFlags.Instance);
```

But I get this:

> REFL018 Bar is explicitly implemented.

**Comment by jnm2** [comment]
Oops, sorry. I left out the binding flags. Also, the class must be defined in an external binary. I can't repro between projects.

This makes sense if the analyzer is not aware of internal members.

**Comment by jnm2** [comment]
The original repro was `typeof(Foo).GetMethod("add_Bar", BindingFlags.NonPublic | BindingFlags.Instance)` but the same error was generated when I used `GetEvent`.

**Comment by jnm2** [comment]
The erroneous `Bar is explicitly implemented.` is actually still happening at the same place in 0.1.18-dev, but I imagine the fix for https://github.com/DotNetAnalyzers/ReflectionAnalyzers/issues/189 will also fix this one.

**Comment by jnm2** [comment]
Hey, this one is actually fixed! I forgot to revert the `BindingFlags.Public` created by https://github.com/DotNetAnalyzers/ReflectionAnalyzers/issues/189 on the same line. 😞 Sorry!

The error makes sense for `BindingFlags.Public` and disappears as expected for `NonPublic`.
